### ISSUE TITLE
Improve contribution guide

### ISSUE THREAD

**Issue by cds-amal** [trufflesuite/truffle#4181]
To encourage our community to participate, our [contribution guide](https://github.com/trufflesuite/truffle/blob/develop/CONTRIBUTING.md) should better document 
  - how we prioritize issues
  - give a better overview of the contribution process
  - explain how to find Issues

**Comment by tansaku** [comment]
would it make sense to describe the develop/master relationship in the contribution guide? e.g. that PRs should go into develop?

**Comment by cds-amal** [comment]
@tansaku yes, absolutely we should describe the git workflow. I updated a rough outline as tasks in the original content. Is there something missing you'd like to see?

**Comment by tansaku** [comment]
maybe the contributing page could start with a little on the truffle mission?  or a link to the mission.  I mean I'm using truffle to build some ethereum smart contract tokens, using it for migrate, console etc. but I'm a little fuzzy on how broad the project is or isn't.  It's great btw, I'm finding it really useful to get stuff deployed, test it out etc.

**Comment by LuisUrrutia** [comment]
I would like to see an improved version of the contribution guide, specifically about compiling, debugging & tools. I found an issue, and I would like to contribute, but I don't even know how to "run" truffle from the repository.

**Comment by cds-amal** [comment]
Hey @LuisUrrutia , are you still interested in contributing? Let me know, and I'll go over the tasks.
